### DISCUSSION TITLE
Coloring borders

### DISCUSSION THREAD

**Discussion by mavam** [ArthurSonzogni/FTXUI#563] [Q&A] [answered]
Is there a way to colorize borders? Adding `| border` alone doesn't give the desired control. I'm catching myself now doing things like `separator() | color(...)`, but that workaround only works in rare occasions. Have you ever considered adding color parameter to `border`?

Here's an example screenshot of a data analytics TUI I'm building:

<img width="1614" alt="image" src="https://user-images.githubusercontent.com/53797/213917775-c8b2713c-429b-4b60-912a-2b5cb24ffcc3.png">

**Comment by ArthurSonzogni** [comment]
Sounds like a good idea to me!

**Reply by mavam** [reply]
What I noticed is that the the current border functions allow for passing in a style (light, single, double), but not a color. If we encode the style in the name of the border function (e.g., `border(Light|Single|Double|Heavy)`, then the color can become an optional argument.

**Reply by ArthurSonzogni** [reply, marked as answer]
Feature added here: https://github.com/ArthurSonzogni/FTXUI/pull/595/files

**Comment by mavam** [comment]
Related question: is there a way to configure the color of a border of a resizable split? I posted [a video](https://asciinema.org/a/LuTCo9XT26h4pwlhdRCvgYPEk) in #632 that uses colored borders, but I cannot apply the same coloring to the resizable split border.

**Reply by ArthurSonzogni** [reply]
After this patch: https://github.com/ArthurSonzogni/FTXUI/commit/9b074d1e27a9fa069d55dee23d75fd8ce4df885a

You should be able to put "arbitrary" element as a separator. For your use case:
```cpp
  auto component = ResizableSplit({
      .main = BasicComponent(),
      .back = BasicComponent(),
      .direction = Direction::Right,
      .main_size = &position,
      .separator_func = [] { return separator() | color(Color::Red) },
  });
```

**Reply by mavam** [reply]
💯
